### PR TITLE
Fix integration test.

### PR DESCRIPTION
**Pull request by fisx**
This was broken in commit 72fefad91e6d.  The spar integration tests now also need to call the new end-point introduced in 68b4db080e68.

## Checklist

 - [x] Title of this PR explains the impact of the change.
 - [x] The description provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] The CHANGELOG.md file in the *Unreleased* section has been updated to explain the change which will be included in the release notes.
 - [x] If a component uses a new or changed internal endpoint of another component, this is mentioned in the CHANGELOG.md.
- [x] If this PR creates a new endpoint, or adds a new configuration flag, the endpoint / config-flag checklist (see Wire-employee only backend [wiki page](https://github.com/zinfra/backend-wiki/wiki/Checklists)) has been followed.

### CODE REVIEW

**Comment by fisx** [comment]
I have no theory as to why this hasn't been caught by the integration tests before merging, though.

**Review by smatting — APPROVED** [review]

**Comment by fisx** [comment]
This is very ugly code, but I really don't want to put a band-aid on it and instead rather start writing a test helper library based on servant-client-generatd functions.  Soon...

**Comment by fisx** [comment]
all tests pass locally.

**Comment by fisx** [comment]
Tests pass locally, but on CI:

```  test-integration/Test/Spar/Scim/UserSpec.hs:1686:24:
  1) Spar.Scim.User, SCIM-managed users, cannot manually update their email, handle or name
       uncaught exception: ErrorCall
       Assertions failed:
        1: 200 =/= 403

       Response was:

       Response {responseStatus = Status {statusCode = 403, statusMessage = "Forbidden"}, responseVersion = HTTP/1.1, responseHeaders = [("Server","nginx"),("Date","Wed, 14 Jul 2021 12:08:31 GMT"),("Content-Type","application/json"),("Transfer-Encoding","chunked"),("Connection","keep-alive"),("Content-Encoding","gzip"),("Access-Control-Allow-Credentials","true"),("Access-Control-Expose-Headers","Request-Id, Location"),("Request-Id","44e9e6acffdf59927aa6462ff0337a06"),("Strict-Transport-Security","max-age=31536000; preload")], responseBody = Just "{\"code\":403,\"message\":\"Missing cookie\",\"label\":\"invalid-credentials\"}", responseCookieJar = CJ {expose = [Cookie {cookie_name = "zuid", cookie_value = "oo-j-5GMBf_aozDdBkv8yefmYS9QKtklVna1VXxG2G5QGtzfCTwYNArrVH0wzGph7i2IAuj8foHn24LpQorjAg==.v=1.k=1.d=1626264631.t=u.l=.u=7c92b7a6-8cd8-447f-be4c-e6308b1e27e5.r=41eb60bc", cookie_expiry_time = 2021-07-14 12:10:31 UTC, cookie_domain = "spar", cookie_path = "/access", cookie_creation_time = 2021-07-14 12:08:31.510863734 UTC, cookie_last_access_time = 2021-07-14 12:08:31.510863734 UTC, cookie_persistent = True, cookie_host_only = True, cookie_secure_only = False, cookie_http_only = True}]}, responseClose' = ResponseClose}
       CallStack (from HasCallStack):
         error, called at src/Bilge/Assert.hs:89:5 in bilge-0.22.0-KwPXBVYHFlF3OFbmYHfwv3:Bilge.Assert
         <!!, called at test-integration/Test/Spar/Scim/UserSpec.hs:1686:24 in main:Test.Spar.Scim.UserSpec
```

It's unlikely to be a timing issue: the error is about the form a request that was constructed in the test, nothing iike a lookup of a user that should be there, but would take 20 more ms to materialize.

The only difference I can think of is that locally, nginz and spar-integration are networked through localhost, and on concourse there is a k8s cluster and at least an haproxy in between them.

**Comment by fisx** [comment]
Or something wrong with the request, and the problem does not materialize on some network setups for some reason?

**Comment by fisx** [comment]
> Or something wrong with the request, and the problem does not materialize on some network setups for some reason?

it was the request: fixed here: 2255828

**Comment by fisx** [comment]
tl;dr: i did make a series of mistakes:

- added nginz to galley integration yaml, but needed it in brig.
- also, the problem wasn't nginz, calling brig right away works exactly as well on `/access*`.
- i used `cookie` from http-client, which uses the cookie jar and only posts cookies with matching domain, but we have a mismatch somewhere in our CI setup.
- i used `header "Cookie"` to construct my own cookie manually, but only in one place, but there were two.

let's see if this is the entire list?
